### PR TITLE
[jit] Allow ScriptModule(optimize=False) when jit disabled

### LINKED ISSUE #16285: [JIT] ScriptModule is not able to take optimize=False as a flag when you have PYTORCH_JIT=0
## 🐛 Bug

ScriptModule is not able to take optimize=False as a flag when you have PYTORCH_JIT=0

## To Reproduce
```
import torch
class ConstantTensor(torch.jit.ScriptModule):
    def __init__(self):
        super(ConstantTensor, self).__init__(optimize=False)
```

This breaks as soon as I use PYTORCH_JIT=0. It should just ignore the kwarg. (rather than breaking)

### PR DESCRIPTION
**Pull request by driazati**
Fixes #16285

### CODE REVIEW

**Comment by eellison** [comment]
Can you add a test ?

File: test/test_jit.py [hunk 1]
```diff
@@ -2224,6 +2224,28 @@ def forward(self, input):
             with self.assertRaisesRegex(RuntimeError, "is a zip"):
                 torch.load(f.name)
 
+    @unittest.skipIf(not PY35, "Python 3.5 needed")
```
Comment (eellison) [inline test/test_jit.py]: Why is Python 3.5 needed ? also might be good to assert that the initial PYTORCH_JIT == PYTORCH_JIT after reloading

**Review by eellison — APPROVED** [review]

File: test/test_jit.py (hunk 1 above)
Comment (ezyang) [reply test/test_jit.py]: Usually better to feature test than to hard code a specific Python version. Intent is clearer too.

Comment (driazati) [reply test/test_jit.py]: The latest commit removes the test, the change is pretty small and the test messing with environment variables / reloading modules is messing up the other tests.

**Pull request closed without merging**